simple-schema.js

```diff
@@ -835,11 +835,14 @@ SimpleSchema.prototype.getDefinition = function(key, propList, functionContext)
 
   // For any options that support specifying a function,
   // evaluate the functions.
-  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'optional', 'label'], function (prop) {
+  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'label'], function (prop) {
     if (_.isFunction(defs[prop])) {
       defs[prop] = defs[prop].call(functionContext || {});
     }
   });
+  if (functionContext && _.isFunction(defs['optional'])) {
+    defs['optional'] = defs['optional'].call(functionContext || {});
+  }
```

Comment:
I fixed the issue regarding context creation.

I didn't change how `SimpleSchema.prototype.getDefinition` is implemented since I think there is no breaking change if I leave it as it is - because `messageForError` doesn't use `optional` prop and `messageForError` is the only function that calls `getDefinition` without context. Of course just assuming all this is true in `getDefinition` function doesn't make this code any more loosely coupled :)

If I call all of these options (`['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'label']`) only when context is defined `messageForError` won't work correctly.

I am not familiar with code enough to refactor how `messageForError` is implemented.